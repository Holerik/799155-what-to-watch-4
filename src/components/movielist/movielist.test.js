// movielist.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Movielist from '../movielist/movielist.jsx';

const filmsInfo = [
  {
    id: 1,
    title: `Joker`,
    poster: `img/joker.jpg`,
    altPoster: `Joker poster`,
    background: `img/joker-bg.jpg`,
    altBackground: `Gotham City`,
    genre: [`Thriller`, `Crime`, `Drama`],
    year: 2019,
    duration: `2h 2min`,
    age: `18+`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 2,
    title: `The Commuter`,
    poster: `img/commuter.jpg`,
    altPoster: `The Commuter poster`,
    background: `img/commuter-bg.jpg`,
    altBackground: `New York City`,
    genre: [`Thriller`, `Action`, `Mystery`],
    year: 2018,
    duration: `1h 45min`,
    age: `16+`,
    src: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
    preview: `https://media.w3.org/2010/05/sintel/trailer.mp4`,
  },
  {
    id: 3,
    title: `Molly's Game`,
    poster: `img/mollys-game.jpg`,
    altPoster: `Molly game poster`,
    background: `img/mollys-game-bg.jpg`,
    altBackground: `New York City`,
    genre: [`Drama`, `Crime`, `Biography`],
    year: 2017,
    duration: `2h 20min`,
    age: `18+`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];

const onMovieTitleClickHandler = jest.fn();

describe(`Movielist tests`, () => {
  it(`Movielist should render several moviecards`, () => {
    const tree = renderer
    .create(
        <Movielist
          filmsInfo={filmsInfo}
          onMovieTitleClick={onMovieTitleClickHandler}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
