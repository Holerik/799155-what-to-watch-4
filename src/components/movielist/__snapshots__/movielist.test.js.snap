// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movielist tests Movielist should render several moviecards 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <img
        alt="Joker poster"
        height="175"
        src="img/joker.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href=""
        onClick={[Function]}
      >
        Joker
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <img
        alt="The Commuter poster"
        height="175"
        src="img/commuter.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href=""
        onClick={[Function]}
      >
        The Commuter
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <img
        alt="Molly game poster"
        height="175"
        src="img/mollys-game.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href=""
        onClick={[Function]}
      >
        Molly's Game
      </a>
    </h3>
  </article>
</div>
`;
